modules/news/index.js

```diff
@@ -55,44 +57,44 @@
               newsArticle.id);
           });
 
-          Promise.all(imagePromises)
-            .then(imageResponses => {
-              var result = _.cloneDeep(news);
+          var results = _.cloneDeep(news);
 
-              for (var i = 0, l = result.length; i < l; i++) {
-                var imageResponse = imageResponses[i];
-                var basePath = this.parent.basePath;
-                var organizationId = this.parent.organizationId;
-                var newsArticleId = result[i].id;
-                var imageSrc = null;
-                if (imageResponse.length) {
-                  imageSrc = util.format('%s/organizations/%s/news/%s/images/%s/data',
-                    basePath,
-                    organizationId,
-                    newsArticleId,
-                    imageResponse[0].id);
-                }
-
-                result[i].imageSrc = imageSrc;
+          async.eachOf(results, (result, index, callback) => {
+            var imagePromise = imagePromises[index];
+            imagePromise.then((imageResponse) => {
+              var basePath = this.parent.basePath;
+              var organizationId = this.parent.organizationId;
+              var newsArticleId = result.id;
+              var imageSrc = null;
+              if (imageResponse.length) {
+                imageSrc = util.format('%s/organizations/%s/news/%s/images/%s/data',
+                  basePath,
+                  organizationId,
+                  newsArticleId,
+                  imageResponse[0].id);
               }
 
-              resolve(result);
-            })
-            .catch(imagesErr => {
-              reject(imagesErr);
+              result.imageSrc = imageSrc;
+              callback();
+            }).catch((imageError) => {
+              console.error('Error loading news image', imageError);
+              callback();
             });
-        })
-          .catch(listErr => {
-            reject(listErr);
+
+          }, (err) => {
```

Comment:
will always be null, probably should delete whole err parameter from last callback